lib/services/ngsi/ngsiService.js

```diff
@@ -208,16 +293,122 @@ function addTimestamp(payload) {
+        if (payload.hasOwnProperty(key)) {
+            addMetadata(payload[key]);
+        }
+    }
+
+    payload[constants.TIMESTAMP_ATTRIBUTE] = timestamp;
+
+    return payload;
+}
+
 /**
  * Makes an update in the Device's entity in the context broker, with the values given in the 'attributes' array. This
- * array should comply to the NGSI's attribute format.
+ * array should comply to the NGSI20's attribute format.
  *
  * @param {String} entityName         Name of the entity to register.
  * @param {Array} attributes        Attribute array containing the values to update.
  * @param {Object} typeInformation  Configuration information for the device.
  * @param {String} token            User token to identify against the PEP Proxies (optional).
  */
-function sendUpdateValue(entityName, attributes, typeInformation, token, callback) {
+function sendUpdateValueNgsi20(entityName, attributes, typeInformation, token, callback) {
+    var payload = {};
+
+    var options = createRequestObject(
+        '/v2/entities/' + entityName + '/attrs',
+        typeInformation,
+        token);
+
+    if (typeInformation && typeInformation.staticAttributes) {
+        attributes = attributes.concat(typeInformation.staticAttributes);
+    }
+
+    if (!typeInformation || !typeInformation.type) {
+        callback(new errors.TypeNotFound(null, entityName));
+        return;
+    }
+
+    for (var i = 0; i < attributes.length; i++) {
+        if (attributes[i].name && attributes[i].type) {
+            payload[attributes[i].name] = {
+                'value' : attributes[i].value,
+                'type' : attributes[i].type,
+                'metadata' : attributes[i].metadata
+            };
+        } else {
+            callback(new errors.BadRequest(null, entityName));
+            return;
+        }
+    }
+
+    async.waterfall([
+        apply(statsService.add, 'measureRequests', 1),
+        apply(applyMiddlewares, updateMiddleware, payload, typeInformation)
+    ], function(error, result) {
+        if (error) {
+            callback(error);
+        } else {
+            if (result) {
+                options.json = result;
+            } else {
+                options.json = payload;
+            }
+
+            if (config.getConfig().timestamp && !utils.isTimestampedNgsi2(options.json)) {
+                options.json = addTimestampNgsi2(options.json);
+            }
+
+            logger.debug(context, 'Updating device value in the Context Broker at [%s]', options.url);
+            logger.debug(context, 'Using the following request:\n\n%s\n\n', JSON.stringify(options, null, 4));
+
+            request(options,
+                generateNGSI2OperationHandler('update', entityName, typeInformation, token, options, callback));
+        }
+    });
+}
+
+/**
+ * Makes an update in the Device's entity in the context broker, with the values given in the 'attributes' array. This
+ * array should comply to the NGSI10's attribute format.
```

Comment:
NGSI10 -> NGSIv1

(In fact NGSI10 and NGSI9 is an old way of naming the two parts of the API... experience has shown is was a bit confusing, so I think is better to try to avoid that naming as possible)

Reply:
Done